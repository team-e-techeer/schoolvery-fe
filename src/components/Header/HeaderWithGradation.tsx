import colors from '@/constants/colors';
import { css } from '@emotion/react';

interface Props {
  schoolName: string;
  imgSrc: string;
  categoryName: string;
}

export default function HeaderWithGradient({ schoolName, imgSrc, categoryName }: Props) {
  return (
    <>
      <header
        css={css`
          height: 17rem;
          background: ${colors.mainColor};
          background: linear-gradient(0deg, #f5b7b7 0%, ${colors.mainColor} 100%);
          padding-left: 5%;
        `}
      >
        <h1
          css={css`
            padding-top: 1rem;
            color: ${colors.grey100};
            font-size: 2.5rem;
          `}
        >
          스쿨버리
        </h1>
        <h2
          css={css`
            color: ${colors.grey100};
            font-size: 1.5rem;
            margin-bottom: 1rem;
          `}
        >
          {schoolName} 대학교
        </h2>
        <div
          css={css`
            display: flex;
            flex-direction: column;
            background-color: ${colors.grey100};
            width: 6rem;
            height: 7rem;
            border-radius: 8px;
          `}
        >
          <img
            css={css`
              width: 6rem;
              height: 6rem;
              object-fit: fill;
            `}
            alt="category-img"
            src={imgSrc}
          />
          <span
            css={css`
              display: flex;
              text-align: center;
              align-content: center;
              bottom: 10%;
              justify-content: center;
              font-size: 1rem;
            `}
          >
            {categoryName}
          </span>
        </div>
      </header>
    </>
  );
}
