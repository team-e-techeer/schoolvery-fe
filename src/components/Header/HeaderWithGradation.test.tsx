import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import HeaderWithGradation from './HeaderWithGradation';
import imgPath from '../../assets/img/dog.png';

describe('로그인 페이지 테스트', () => {
  it('렌더링 테스트', () => {
    const { container } = render(<HeaderWithGradation schoolName="한국공과" categoryName="강아지" imgSrc={imgPath} />, {
      wrapper: BrowserRouter,
    });
    expect(screen.getByText('스쿨버리')).toBeInTheDocument();
    expect(screen.getByText('한국공과 대학교')).toBeInTheDocument();
    expect(screen.getByAltText('category-img')).toBeInTheDocument();
    expect(screen.getByText('강아지')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
