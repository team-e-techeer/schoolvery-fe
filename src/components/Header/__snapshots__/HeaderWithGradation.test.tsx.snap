// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`로그인 페이지 테스트 렌더링 테스트 1`] = `
.emotion-0 {
  height: 17rem;
  background: rgb(249, 168, 168);
  background: linear-gradient(0deg, rgba(249, 168, 168, 1) 0%, rgba(91, 52, 170, 1) 100%);
  padding-left: 5%;
}

.emotion-1 {
  padding-top: 1rem;
  color: #f2f4f6;
  font-size: 2.5rem;
}

.emotion-2 {
  color: #f2f4f6;
  font-size: 1.5rem;
  margin-bottom: 1rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #f2f4f6;
  width: 6rem;
  height: 7rem;
  border-radius: 8px;
}

.emotion-4 {
  width: 6rem;
  height: 6rem;
  object-fit: fill;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  bottom: 10%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 1rem;
}

<div>
  <header
    class="emotion-0"
  >
    <h1
      class="emotion-1"
    >
      스쿨버리
    </h1>
    <h2
      class="emotion-2"
    >
      한국공과
       대학교
    </h2>
    <div
      class="emotion-3"
    >
      <img
        alt="category-img"
        class="emotion-4"
        src="test-file-stub"
      />
      <span
        class="emotion-5"
      >
        강아지
      </span>
    </div>
  </header>
</div>
`;
